src/kartograf/atom_mapper.py

```diff
@@ -873,10 +904,46 @@ def suggest_mappings(
         Iterator[AtomMapping]
             returns an interator of possible atom mappings.
         """
-        yield LigandAtomMapping(
-            A,
-            B,
-            self.suggest_mapping_from_rdmols(
-                molA=A.to_rdkit(), molB=B.to_rdkit()
-            ),
-        )
+        # TODO: We probably want to modularize the following in methods
+        if isinstance(A, ProteinComponent) or isinstance(B, ProteinComponent):
+            # 1. identify Component Chains
+            componentA_chains = self._split_protein_components_molecules(A)
+            componentB_chains = self._split_protein_components_molecules(B)
+
+            # 2. calculate all possible mappings
+            largest_mappings = []
+            for A_chain in componentA_chains:
+                largest_overlap_map = {}  # Initialize to empty map
+                largest_overlap_component = componentB_chains[0]  # Initialization
+                for B_chain in componentB_chains:
+                    # This reinitializes indices, that's why we need the index information from
+                    #  split components.
+                    current_map = self.suggest_mapping_from_rdmols(
+                        molA=A_chain.to_rdkit(), molB=B_chain.to_rdkit()
+                    )
+                    if len(largest_overlap_map) < len(current_map):
+                        largest_overlap_component = B_chain
+                        largest_overlap_map = current_map
+                # TODO: Do we need a better suited object here instead of LigandAtomMapping?
+                mapping_obj = LigandAtomMapping(A_chain, largest_overlap_component, largest_overlap_map)
+                # At the end of the loop mapping_obj should have the largest map overlap
+                largest_mappings.append(mapping_obj)
+
+            # Merge all the largest mappings for each component into a single mapping
+            merged_map = {}
+            for mapping_obj in largest_mappings:
+                start_a = int(mapping_obj.componentA.name.split("_")[-1])
+                start_b = int(mapping_obj.componentB.name.split("_")[-1])
+                shifted_map = {a_idx + start_a: b_idx + start_b for a_idx, b_idx in
```

Comment:
Just to note here that we have a little bit of a footgun here,  when we shift the indices and we update the dictionary it is possible that some of the indices get overwritten and that means that probably something went wrong. 

I don't know what's a good solution for this, but maybe we should think about having yet another class that handles this itself, maybe inheriting from `dict` and throwing an exception when a `__setitem__` overwrites something that already exists. Just a guess at this.

Reply:
Even at the expense of non-fancy code & extra cost, it might be good to do a check on the indices and make sure that there's no dupllicates. In its simplest form, just a loop where you create the two lists, turn them into sets and see if the length changed?